lib/google_maps.ex

```diff
@@ -320,27 +321,41 @@ defmodule GoogleMaps do
 
   @spec distance(address(), address(), options()) :: Response.t()
   def distance(origin, destination, options) when is_binary(origin) and is_binary(destination) do
-    params = options
-    |> Keyword.merge([origins: origin, destinations: destination])
+    params =
+      options
+      |> Keyword.merge(origins: origin, destinations: destination)
 
     GoogleMaps.get("distancematrix", params)
   end
 
+  @spec distance([address()], [address()], options()) :: Response.t()
+  def distance([origin | _] = origins, [destination | _] = destinations, options)
+      when is_binary(origin) and is_binary(destination) do
```

Comment:
Yeah thats a rock solid idea. I ended up changing the spec to be [coordinate() | address()] because I thought it added a bit more clarity as opposed to merging the types. Let me know what you think.